pkg/asset/installconfig/gcp/services.go

```diff
@@ -4,81 +4,97 @@ import (
+	storageServiceNameGCP   serviceNameGCP = "storage"
+
+	serviceVersionGCPv1 serviceVersionGCP = "v1"
 )
 
+// FormatGCPEndpointInput is the structure containing input variables for formatting the GCP Service Endpoints.
+type FormatGCPEndpointInput struct {
+	// SkipPath should be set to true when the path should not be added to the
+	// formatted endpoint. When the path is added, an example endpoint of
+	// https://compute-exampleendpoint.p.googleapis.com would be formatted as
+	// https://compute-exampleendpoint.p.googleapis.com/compute/v1/
+	SkipPath bool
+}
+
 // FormatGCPEndpointList will format the list of GCP Service Endpoints to match the expected url
-// for WithEndpoint or BasePath override endpoint options
-func FormatGCPEndpointList(endpoints []configv1.GCPServiceEndpoint) ([]configv1.GCPServiceEndpoint, error) {
+// for WithEndpoint or BasePath override endpoint options.
+func FormatGCPEndpointList(endpoints []configv1.GCPServiceEndpoint, input FormatGCPEndpointInput) ([]configv1.GCPServiceEndpoint, error) {
 	// The endpoints are modified to include the path
 	modifiedEndpoints := []configv1.GCPServiceEndpoint{}
 	for _, se := range endpoints {
-		var err error
-		var formattedURL string
-		switch se.Name {
-		case configv1.GCPServiceEndpointNameCloudResource:
-			formattedURL, err = FormatGCPEndpoint(se.URL, CloudResourceManagerNameGCP, ServiceVersionGCP3)
-		case configv1.GCPServiceEndpointNameCompute:
-			formattedURL, err = FormatGCPEndpoint(se.URL, ComputeServiceNameGCP, ServiceVersionGCP1)
-		case configv1.GCPServiceEndpointNameDNS:
-			formattedURL, err = FormatGCPEndpoint(se.URL, DNSServiceNameGCP, ServiceVersionGCP1)
-		//case configv1.GCPServiceEndpointNameFile:
-		//	formattedURL, err = FormatGCPEndpoint(se.URL, FileServiceNameGCP, ServiceVersionGCP1)
-		case configv1.GCPServiceEndpointNameIAM:
-			formattedURL, err = FormatGCPEndpoint(se.URL, IAMServiceNameGCP, ServiceVersionGCP1)
-		//case configv1.GCPServiceEndpointNameServiceUsage:
-		//	formattedURL, err = FormatGCPEndpoint(se.URL, ServiceUsageNameGCP, ServiceVersionGCPBeta)
-		case configv1.GCPServiceEndpointNameStorage:
-			formattedURL, err = FormatGCPEndpoint(se.URL, StorageServiceNameGCP, ServiceVersionGCP1)
+		formattedURL, err := FormatGCPEndpoint(se.Name, se.URL, input)
+		if err != nil {
+			return nil, fmt.Errorf("failed to format GCP service endpoint %s: %w", se.Name, err)
 		}
 
-		if err != nil || formattedURL == "" {
-			return nil, fmt.Errorf("failed to format GCP Service Endpoint URL %s: %w", se.URL, err)
-		}
+		logrus.Debugf("Formatted GCP service endpoint %s: %s", se.Name, formattedURL)
 		modifiedEndpoints = append(modifiedEndpoints, configv1.GCPServiceEndpoint{Name: se.Name, URL: formattedURL})
 	}
 	return modifiedEndpoints, nil
 }
 
 // FormatGCPEndpoint will format the endpoint to ensure that the string is in the format that would be
 // accepted by both options (WithEndpoint and BasePath override).
```

Comment:
Would be good to add examples of what the WithEndpoint and BasePath override options mean.